content/knowledge/scalinghistory.md

```diff
@@ -0,0 +1,135 @@
+
+In theory, PoW can scale infinitely.
+But with it comes the extra overhead of computational complexity which may not be practical in a truly global sense.
+Also some PoW based blockchains like Ethereum1.0 suffer from huge transaction fees as well as low throughput of 10-15 transactions per second which make them impractical for true global usage.
+   Therefore the PoW consensus based blockchains can achieve decentralisation and security but not scalability.
+
+## 2. Federated Byzantine Agreement (FBA)
+
+   After Bitcoin and PoW came into picture in 2008, another class of consensus protocols surfaced around 2014 which tried to tackle a different problem known as “Byzantine Generals Problem”.
+   This is the problem when there are various nodes communicating with each other and a few of them are “Byzantine” which means either faulty or malicious.
+   A good consensus protocol should be able to tackle this issue in a decentralised manner for better security of the network.
+   One of the consensus algorithms that manage to do it is called “Federated Byzantine Agreement”.
+   This was introduced by Stellar.
+   In FBA, every node has its “quorum set” which is a list of other nodes which this particular node trusts.
+   Therefore to achieve consensus for a transaction, the node relies on the members in its quorum set.
+   As long as the majority of nodes are not malicious, this will ensure security of the system.
+   Complex web of overlapping quorum slices makes it almost impossible for the majority of nodes to collude to control consensus.
+   FBA can be used on top of PoW or PoS (to be discussed later).
+   Safety and fault tolerance are [preferred over liveness](https://www.youtube.com/watch?v=aU08km2xrz0&ab_channel=Lumenauts).
+   In case of an accidental fork, the system is halted until consensus is reached.
+   This is important in case of banking applications.
+   Consensus that requires only message passing followed by voting process leads to high transaction volume per second and less expenditure of electricity than PoW.
+   However, a criticism of pure FBA is that it leads to fragile structures of constituent nodes, permitting topology scenarios where a single node failure can cause a network-wide failure.
+
+   TODO: Drawback: Single point failure if quorum sets share a single node?
+   (Not mentioning about BFT which require a pre recommended validator list as it is against decentralisation)
+
+## 3. Proof of Stake (PoS)
+
+   Then came the Proof of Stake(PoS) consensus algorithms (e.g. Casper).
+   This no longer needs the computational complexity for validation of blocks but rather how much a node or node pool has staked and how long it has been active.
+   Staking is like putting some native cryptocurrency as a collateral.
+   The higher the stake and time in the game, the higher the chances of getting to write a block on behalf of the whole ecosystem and in return receive some cryptocurrency.
```

Comment:
Time in the game is not relevant for most blockchains. You usually have staking periods (epochs). If you want to start validating, to commit a stake; after the epoch changeover, your stake is committed (illiquid) and you can start validating. If you want to stop validating and use your funds, you request unstaking, which usually takes at least one complete epoch as well.